src/components/molecules/ScheduleItemNG/ScheduleItemNG.tsx

```diff
@@ -74,31 +71,29 @@ export const ScheduleItemNG: React.FC<ScheduleItemNGProps> = ({
     differenceInCalendarDays(eventEndTime(event), eventStartTime(event))
   );
   const isCurrentEventLive = isEventLive(event);
-  const roomUrlParam = getUrlParamFromString(event.room ?? "");
 
   const handleCopyEventLink = useCallback(
     (e?: React.MouseEvent<HTMLButtonElement>) => {
+      // @debt get rid of stopPropagation() in the project allowing a valid event bubbling
       e && e.stopPropagation();
 
-      const eventLink = getFullVenueInsideUrl(roomUrlParam);
+      const eventLink = eventRoom
+        ? eventRoom.url
+        : getFullVenueInsideUrl(eventVenue?.id ?? "");
       navigator.clipboard.writeText(eventLink);
+      setIsEventLinkCopied(true);
+      setTimeout(setIsEventLinkCopied, 1000, false);
```

Comment:
Hacky depending on a 1 second delay to get it working.
Proper code should use promises to wait for a job done signal.

Also,

Good JS practice is to always use ()=>{} function instead of relying on `setTimeout` to pass parameters for you. I just can't remember when was the last time I saw an example why that came to be.

```suggestion
      setTimeout( ()=>setIsEventLinkCopied(false), 1000);
```


[OOS]

Could use a delay execute function, maybe even some library has one.

Reply:
it doesn't wait for clipboard to copy. It just changes the text on the button back from "Copied!" to "Copy to Clipboard"
![recording](https://user-images.githubusercontent.com/63313732/131528976-f26c21dc-5351-4874-acb6-e279ee968ddd.gif)

() => foo() -> fixed https://github.com/sparkletown/sparkle/pull/2183/commits/2be26c4e66c5595df4c893f818cc364b297a21b3